src/goto-checker/multi_path_symex_checker.cpp

```diff
@@ -155,3 +166,19 @@ multi_path_symex_checkert::localize_fault(const irep_idt &property_id) const
 
   return fault_localizer(property_id);
 }
+
+void multi_path_symex_checkert::report()
+{
+  if(options.is_set("write-solver-stats-to"))
+  {
+    if(
+      auto hardness_collector =
+        dynamic_cast<hardness_collectort *>(&property_decider.get_prop()))
```

Comment:
See below. `propt` must not be exposed. Forward to `prop_conv_solvert` which is a `decision_proceduret`.